templates/dart/lib/client.dart.twig

```diff
@@ -8,49 +8,42 @@ class Client {
     String endPoint;
     Map<String, String> headers;
     bool selfSigned;
-    Dio http;
-    
-    Client() {
-        this.endPoint = '{{spec.endpoint}}';
+    final Dio http;
+
+    Client({this.endPoint: 'https://appwrite.io/v1', this.selfSigned: false, Dio http}) : this.http = http ?? Dio() {
         this.headers = {
             'content-type': 'application/json',
             'x-sdk-version': '{{spec.title | caseDash}}:{{ language.name | caseLower }}:{{ sdk.version }}',
         };
-        this.selfSigned = false;
 
-        this.http = Dio();
+        assert(endPoint.startsWith(RegExp("http://|https://")), "endPoint $endPoint must start with 'http'");
```

Comment:
Is this method related to a unit-tests library?

Reply:
No, it's not. But I've decided to dive deeper into docs and find out that assertions are not used in production code. So if developer makes a mistake, he will see an error in logs during development. But it won't be evaluated on end users phones.

Reply:
👌🏻